src/server_manager/web_app/app.ts

```diff
@@ -121,6 +122,23 @@ async function showHelpBubblesOnce(serverView: Polymer) {
   }
 }
 
+// Displays a survey dialog with title `surveyTitle` and a link to `surveyLink`.
+// Does not display the survey with `surveyId` if it has already been shown to
+// the user or if the current date is after `displayBefore`.
+async function showHatsDialogOnce(
+    hatsDialog: Polymer, surveyId: string, surveyTitle: string, surveyLink: string,
+    displayBefore?: Date) {
+  const now = new Date();
+  if (displayBefore && now > displayBefore) {
```

Comment:
The display before is survey-specific. It seems it belongs somewhere else, not in a generic Hats code.

Reply:
@cjhenck, do you anticipate this requirement applying to other features?